### PR TITLE
Send project metadata to the index

### PR DESCRIPTION
**Pull request by tmeasday**
Reverts chromaui/chromatic-cli#1121 (re-implements https://github.com/chromaui/chromatic-cli/pull/1112) and incorporates https://github.com/chromaui/chromatic-cli/pull/1120.

Also:

 - added some error/timeout handling so we don't get stuck if there is something wrong with git.
 - shorted timeouts to 5s as it's not critical this works (we could shorten further?)
 - don't use `wc` or `head` as they may not be there. Instead:
   - simulate `head -n 1` via streams
   - just count the number of lines for `wc`. 
      - This is inefficient as it requires loading all the lines into memory. 
      - I'm not sure if this will be a problem or not if a repo has a heap of contributors or "app files".  
      - I could implement another `execGitCommandCountLines` but I figured it was better to keep the complexity down.

Things to check in QA:

- environments without access to `head` or `wc`.
- environments where these commands timeout [not sure how to simulate]

<details>
  <summary>📦 Published PR as canary version: <code>11.20.0--canary.1122.12129895074.0</code></summary>
  <br />

  :sparkles: Test out this PR locally via:
  
  ```bash
  npm install chromatic@11.20.0--canary.1122.12129895074.0
  # or 
  yarn add chromatic@11.20.0--canary.1122.12129895074.0
  ```
</details>

### CODE REVIEW

**Comment by jmhobbs** [comment]
> just count the number of lines for wc.
>  This is inefficient as it requires loading all the lines into memory.

I think it's a good trade off to add a `execGitCommandCountLines`.  I'd rather read newlines out of the stream and discard with one additional function than keep unbounded values in memory, twice after the `.split('\n')`

**Comment by tmeasday** [comment]
@jmhobbs done! Looks like it still works (2 and 0 are the correct values for this repo):

image

File: node-src/lib/getDependentStoryFiles.test.ts [hunk 1]
```diff
@@ -2,26 +2,30 @@
...
+  }: { configDir?: string; staticDir?: string } & Context['options'] = {} as any
+) => ({
   log,
-  options: { storybookBaseDir: '.', ...options },
+  options,
   turboSnap: {},
-  storybook: { configDir, staticDir },
-  git: {},
+  storybook: {
+    baseDir: options.storybookBaseDir ? options.storybookBaseDir.replace(/^\.[/\\]?/, '') : '',
```
Comment (jmhobbs) [inline node-src/lib/getDependentStoryFiles.test.ts:24]: Could you throw a comment on here about that regex?

**Review by jmhobbs — APPROVED** [review]
Haven't QA'd, and there's some CI issues, but overall looks right to me.

**Comment by tmeasday** [comment]
@jmhobbs FYI the CI issues (outside of Codacy) is due to the production index not yet having `projectMetadata.storybookBaseDir` supported yet (need to merge the relevant PR on the other end). This is a good thing I guess (we can't merge this until that's done!)

File: node-src/lib/getDependentStoryFiles.test.ts (hunk 1 above)
Comment (tmeasday) [reply node-src/lib/getDependentStoryFiles.test.ts:24]: I got rid of this. I think it's simpler to just pass up the string the user has configured and do any re-mapping on the backend.

**Pull request merged**